### PR TITLE
SALTO-5686: Validate dynamic OS constraints on DeviceAssurancePolicy (Okta)

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Add a validator to notify user when trying to deploy device assurance with dynamic os constraints and the feature is not enabled in the account.

---

_Additional context for reviewer_

---
_Release Notes_: 
_Okta_adapter_:
- When deploying Device Assurance Policies with dynamic OS constraints, verify the dynamic OS feature is enabled in the target account

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/src/change_validators/dynamic_os_version_feature.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const log = logger(module)
+
+const DYNAMIC_OS_VERSION_FEATURE_NAME = 'Dynamic OS version compliance'
+
+const DYNAMIC_OS_VERSION_PATH_OPTIONS = [
+  ['osVersion', 'dynamicVersionRequirement'],
+  ['osVersionConstraints'],
+  // support paths including additionalProperties for backward compatibility
+  ['osVersion', 'additionalProperties', 'dynamicVersionRequirement'],
+  ['additionalProperties', 'osVersionConstraints'],
```
Comment (omrilit) [inline packages/okta-adapter/src/change_validators/dynamic_os_version_feature.ts:32]: nit: Do we want to add a TODO to remove this one and link to the relevant ticket?

File: packages/okta-adapter/src/change_validators/dynamic_os_version_feature.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+export const dynamicOSVersionFeatureValidator: ChangeValidator = async (changes, elementSource) => {
+  if (elementSource === undefined) {
+    log.error('Failed to run dynamicOSVersionFeatureValidator because element source is undefined')
+    return []
+  }
+
+  const relevantDeviceAssuranceInstances = changes
+    .filter(isInstanceChange)
+    .filter(isAdditionOrModificationChange)
+    .map(getChangeData)
```
Comment (omrilit) [inline packages/okta-adapter/src/change_validators/dynamic_os_version_feature.ts:49]: You are looking only on the `after` element.
Is it possible to have a certain value in the `before` that was removed from the `after`?

**Review by omrilit — APPROVED** [review]
Looks great!

File: packages/okta-adapter/src/change_validators/dynamic_os_version_feature.ts (hunk 2 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/src/change_validators/dynamic_os_version_feature.ts:49]: mm not sure I get what you meant.. I only look on the after cause I care a about the values being sent in the API call

**Pull request merged**